cli/types/cypress.d.ts

```diff
@@ -2978,19 +2978,15 @@ declare namespace Cypress {
 
   type DevServerFn<ComponentDevServerOpts = any> = (cypressConfig: DevServerConfig, devServerConfig: ComponentDevServerOpts) => ResolvedDevServerConfig | Promise<ResolvedDevServerConfig>
   interface ComponentConfigOptions<ComponentDevServerOpts = any> extends CoreConfigOptions {
-    // TODO(tim): Keeping optional until we land the implementation
-    devServer?: Promise<{ devServer: DevServerFn<ResolvedDevServerConfig>}> | { devServer: DevServerFn<ResolvedDevServerConfig> } | DevServerFn<ResolvedDevServerConfig>
+    devServer: Promise<{ devServer: DevServerFn<ComponentDevServerOpts>}> | { devServer: DevServerFn<ComponentDevServerOpts> } | DevServerFn<ComponentDevServerOpts>
     devServerConfig?: ComponentDevServerOpts | Promise<ComponentDevServerOpts>
   }
 
   /**
    * Takes ComponentDevServerOpts to track the signature of the devServerConfig for the provided `devServer`,
    * so we have proper completion for `devServerConfig`
    */
-  type ConfigOptions<ComponentDevServerOpts = any> = CoreConfigOptions & {
-    e2e?: CoreConfigOptions,
-    component?: ComponentConfigOptions<ComponentDevServerOpts>
-  }
+  type ConfigOptions<ComponentDevServerOpts = any> = Partial<ResolvedConfigOptions<ComponentDevServerOpts>>
```

Comment:
Any reason this changed to a `Partial<>`?

Reply:
- A resolved config has every default in, so every member is required.
- The config object has nothing required since you could even "not have a file".

Hence the need to have a partial for this.